File: src/main/java/dev/rafi/vitelist/commands/sub/On.java
package dev.rafi.vitelist.commands.sub;

import com.velocitypowered.api.command.CommandSource;
import dev.rafi.vitelist.storage.Config;
import dev.rafi.vitelist.utils.ChatUtils;
import dev.rafi.vitelist.utils.QueryUtils;

public class On {
    public static void whitelistOn(CommandSource executor, String[] args) {
        if (!executor.hasPermission("vitelist.commands.on")) {
            ChatUtils.sendMessage(executor, Config.NO_PERMISSION);
            return;
        }

        if (args.length == 1) {
            QueryUtils.turnOnWhiteList("all");
            ChatUtils.sendMessage(executor, Config.ENABLE);
        } else if (args.length == 2) {
            if (!QueryUtils.serverExists(args[1])) {
                ChatUtils.sendMessage(executor, Config.WRONG_SERVER.replace("%server%", args[1]));
                return;
            }
            QueryUtils.turnOnWhiteList(args[1]);
            ChatUtils.sendMessage(executor, Config.ENABLE_SERVER.replace("%server%", args[1]));
        }
    }
}

File: src/main/kotlin/dev/rafi/vitelist/commands/subcommands/On.kt
package dev.rafi.vitelist.commands.subcommands

import com.velocitypowered.api.command.CommandSource
import dev.rafi.vitelist.database.model.Server
import dev.rafi.vitelist.storage.Config
import dev.rafi.vitelist.utils.sendMessage
import dev.rafi.vitelist.utils.serverExists
import org.jetbrains.exposed.sql.transactions.transaction
import org.jetbrains.exposed.sql.update

fun on(source: CommandSource, args: Array<String>) {
    if (!(source.hasPermission("vitelist.commands.on"))) {
        sendMessage(source, Config.NO_PERMISSION)
        return
    }
    if (args.size == 1) {
        transaction {
            Server.update({ Server.status eq false }) {
                it[status] = true
            }
        }
        sendMessage(source, Config.ENABLE_GLOBAL)
    } else {
        val serverName = args[1]

        if (!serverExists(serverName)) {
            sendMessage(source, Config.SERVER_NOT_FOUND.replace("%server%", serverName))
            return
        }

        transaction {
            Server.update({ Server.name eq serverName }) {
                it[status] = true
            }
        }
        sendMessage(source, Config.ENABLE_SERVER.replace("%server%", serverName))
    }
}